src/components/topo/ucc_sbgp.c

```diff
@@ -203,18 +205,33 @@ static ucc_status_t sbgp_create_node_leaders(ucc_topo_t *topo, ucc_sbgp_t *sbgp,
         return UCC_ERR_NO_MEMORY;
     }
 
+    nl_array_3 = ucc_malloc(comm_size * nnodes * sizeof(ucc_rank_t),
+                            "nl_array_3");
+    if (!nl_array_3) {
+        ucc_error("failed to allocate %zd bytes for nl_array_3",
+                  comm_size * nnodes * sizeof(ucc_rank_t));
+        ucc_free(nl_array_1);
+        ucc_free(nl_array_2);
+        return UCC_ERR_NO_MEMORY;
+    }
+
     for (i = 0; i < nnodes; i++) {
         nl_array_1[i] = 0;
         nl_array_2[i] = UCC_RANK_MAX;
     }
+    for (i = 0; i < nnodes * comm_size; i++) {
+        nl_array_3[i] = 0;
+    }
 
     for (i = 0; i < comm_size; i++) {
-        ucc_rank_t    ctx_rank = ucc_ep_map_eval(set->map, i);
-        ucc_host_id_t host_id  = topo->topo->procs[ctx_rank].host_id;
+        ucc_rank_t      ctx_rank  = ucc_ep_map_eval(set->map, i);
+        ucc_host_id_t   host_id   = topo->topo->procs[ctx_rank].host_id;
+        ucc_socket_id_t socket_id = topo->topo->procs[ctx_rank].socket_id;
```

Comment:
how do we handle case when rank is not socket bound?

Reply:
fixed.